viewing_party/party.py

```diff
@@ -1,23 +1,153 @@
 
+from pyparsing import empty
+
 def create_movie(title, genre, rating):
-    pass
+    # creating a dictionary that stores the variables passed into the function as the values
+    new_movie_added = {
+        "title": title,
+        "genre": genre,
+        "rating": rating
+    }
+    # checks to make sure that the variables passed into the function are real values and not none. 
+    # If none, returns. If valid input, then returns the dict.
+    if title is None or genre is None or rating is None:
+        return 
+    else:
+        return new_movie_added
+
+def add_to_watched(user_data, movie):
+    # adds movies passed as an argument to the user_data["watched"] list 
+    user_data["watched"].append(movie)
+    
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    # adds movies passed as an argument to the user_data["watched"] list 
+    user_data["watchlist"].append(movie)
+    
+    return user_data
+
+def watch_movie(user_data, title):   
+    for movie in range(len(user_data["watchlist"])): # loops through user_data["watchlist"] 
+        if title in user_data["watchlist"][movie]["title"]: # conditional that checks if the title is in the user_data["watchlist"] at index i 
+            user_data["watchlist"].remove(user_data["watchlist"][movie]) # removes the title from the user_data["watchlist"]  at index i 
+            user_data["watched"].append(title) # adds the title to the user_data["watched"] 
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    # initialize list object that will store all movie ratings from user_data["watched"]
+    user_movie_ratings = []
+    # inititalize float variable that stores ratings for empty lists 
+    empty_list_rating = 0.0
+
+    for movie in user_data["watched"]: # loops through movies in "watched"
+        user_movie_ratings.append(movie["rating"]) # adds the ratings from "watched" to ratings_list
+    
+    if len(user_data["watched"]) != 0: # if list contains elements and is not equal to 0
+        # calculates average by dividing the sum of the list with the length of the list
+        movie_ratings_average = sum(user_movie_ratings) / len(user_movie_ratings) 
+    else:
+        return empty_list_rating # returns 0.0
+    
+    return movie_ratings_average 
+
+def get_most_watched_genre(user_data):
+    # initialize list object that will store the genres of all the movies in user_data["watched"]
+    user_movie_genre = []
+
+    for movie in user_data["watched"]: # loops through movie in user_data["watched"] list 
+        user_movie_genre.append(movie["genre"]) # appends values of "genre" key to user_movie_genre list
+
+    if len(user_data["watched"]) != 0: # if the list contains elements and is greater than 0 
+        user_movie_genre.sort() # sorts list in place (modifies original list)

```

Comment:
Good idea to use `.sort()`! Nice call about potentially modifying data; nice to use it on `user_movie_genre` and not modify the input user_data.